org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/colors/utils/ColorUtils.java

```diff
@@ -0,0 +1,374 @@
+				double red = getNumericValue(colorValues[0], 255.0);
+				double green = getNumericValue(colorValues[1], 255.0);
+				double blue = getNumericValue(colorValues[2], 255.0);
+				return new Color(red, green, blue, alpha);
+			}
+			if (text.startsWith("rgb(") || text.startsWith("rgba(")) {
+
+			}
+		} catch (Exception e) {
+			return null;
+		}
+		return null;
+	}
+
+	private static double getNumericValue(String value, double factor) {
+		double result = Double.parseDouble(value);
+		return result / factor;
+	}
+
+	/**
+	 * Returns the RGB color presentation of the given <code>color</code> and
+	 * <code>replace</code> range.
+	 * 
+	 * @param color   the color.
+	 * @param replace the replace range.
+	 * 
+	 * @return the RGB color presentation of the given <code>color</code> and
+	 *         <code>range</code>.
+	 */
+	public static ColorPresentation toRGB(Color color, Range replace) {
+		int red256 = (int) Math.round(color.getRed() * 255);
+		int green256 = (int) Math.round(color.getGreen() * 255);
+		int blue256 = (int) Math.round(color.getBlue() * 255);
+		int alpha = (int) color.getAlpha();
+
+		String label = getRGB(red256, green256, blue256, alpha == 1 ? null : alpha);
+		TextEdit textEdit = new TextEdit(replace, label);
+		return new ColorPresentation(label, textEdit);
+	}
+
+	private static String getRGB(int red256, int green256, int blue256, Integer alpha) {
+		StringBuilder label = new StringBuilder("rgb(");
+		label.append(red256);
+		label.append(",");
+		label.append(green256);
+		label.append(",");
+		label.append(blue256);
+		if (alpha != null) {
+			label.append(",");
+			label.append(alpha);
+		}
+		label.append(")");
+		return label.toString();
+	}
+
+	/**
+	 * Returns the Hexa color presentation of the given <code>color</code> and
+	 * <code>replace</code> range.
+	 * 
+	 * @param color   the color.
+	 * @param replace the replace range.
+	 * 
+	 * @return the Hexa color presentation of the given <code>color</code> and
+	 *         <code>range</code>.
+	 */
+	public static ColorPresentation toHexa(Color color, Range replace) {
+		int red256 = (int) Math.round(color.getRed() * 255);
+		int green256 = (int) Math.round(color.getGreen() * 255);
+		int blue256 = (int) Math.round(color.getBlue() * 255);
+		int alpha = (int) color.getAlpha();
+
+		String label = getHexa(red256, green256, blue256, alpha == 1 ? null : alpha);
+		TextEdit textEdit = new TextEdit(replace, label);
+		return new ColorPresentation(label, textEdit);
+	}
+
+	private static String getHexa(int red256, int green256, int blue256, Integer alpha) {
+		StringBuilder label = new StringBuilder("#");
+		label.append(toTwoDigitHex(red256));
+		label.append(toTwoDigitHex(green256));
+		label.append(toTwoDigitHex(blue256));
+		if (alpha != null) {
+			label.append(toTwoDigitHex(Math.round(alpha * 255)));
+		}
+		return label.toString();
+	}
+
+	private static String toTwoDigitHex(int n) {
+		String r = String.valueOf(n);

```

Comment:
I think the issue with converting between rgb and here comes from here because `valueOf()`  will convert the hex to integer (ex: `0xff` will become `255`, and in the next like will become `0255`), which doesn't work.

If you change this line to `String r = Integer.toHexString(n);` (`toHexString()` will convert `0xff` to `ff`), it should work like this: 

![Peek 2023-01-05 11-11](https://user-images.githubusercontent.com/73968480/210827464-1ca74556-da21-4bf5-bd8f-cafeef1bdebd.gif)